tests/common/utilities.py

```diff
@@ -488,8 +488,15 @@ def compare_crm_facts(left, right):
     for k, v in list(left['resources'].items()):
         lv = v
         rv = right['resources'][k]
-        if lv['available'] != rv['available'] or lv['used'] != rv['used']:
-            unmatched.append({'left': {k: lv}, 'right': {k: rv}})
+
+        if duthost.facts['platform_type'] == "cisco-8000":
+            # For Cisco-8000 devices, hardware counters are statistical-based with +/- 1 entry tolerance.
+            # Hence, the available counter may not increase as per initial value.
+            if abs(int(lv['available']) - int(rv['available'])) > 1 or abs(int(lv['used']) - int(rv['used'])) > 1:
```

Comment:
another option is, instead of adding the check here, get_crm_facts can be collected and the comparison can be done within the calling script itself. Or go with the optional arguments method